### PR TITLE
test(factories): add factory tests

### PR DESCRIPTION
**Pull request by levithomason**
This PR adds tests for `createShorthand`, it does not add tests for every shorthand factory, however, they are partially applied so it is _almost_ redundant.

### CODE REVIEW

File: src/lib/factories.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ const mergePropsAndClassName = (defaultProps, props) => {
     newProps.className = cx(defaultProps.className, props.className) // eslint-disable-line react/prop-types
   }
 
-  if (!newProps.key) {
+  if (!newProps.key && childKey) {
```
Comment (levithomason) [inline src/lib/factories.js:19]: Prevent adding `childKey: undefined` due to props with no key and also not defining a `childKey`.  This issue was revealed by tests.

**Comment by levithomason** [comment]
Just force pushed `from an array` tests

File: test/specs/lib/factories-test.js [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+      })
+    })
+
+    describe('from null', () => {
+      itReturnsNull(null)
+      itReturnsNullGivenDefaultProps(null)
+      itReturnsNullGivenDefaultPropsFunc(null)
+    })
+
+    describe('from undefined', () => {
```
Comment (jeffcarbs) [inline test/specs/lib/factories-test.js]: Add similar tests for booleans. Both `true` and `false` should return `null`

File: test/specs/lib/factories-test.js [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    const props = { 'data-foo': 'foo' }
+    const wrapper = shallow(getShorthand({ value, mapValueToProps: () => props }))
+
+    _.each(props, (val, key) => {
+      wrapper.should.not.have.prop(key, val)
+    })
+  })
+}
+
+const itMergesClassNames = (extraClassName, classNameSource, shorthandConfig) => {
```
Comment (jeffcarbs) [inline test/specs/lib/factories-test.js]: Put `classNameSource` as first arg:
- matches `propsSource` being first in the functions below
- currently args 1 and 3 are used in the test and 2 is used to describe the test which is a little confusing.

**Review by jeffcarbs — APPROVED** [review]
Left a few suggestions but looks good otherwise so I'm approving 👍

**Comment by levithomason** [comment]
Thanks much, updated both suggestions.  Will merge on pass.

**Pull request merged**